lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -2268,20 +2272,117 @@ static void computeReuse(TransferReadOp readOp, VectState *state) {
   }
 }
 
+static LogicalResult isUnalignedLoad(TransferReadOp readOp, VectState *state) {
+  VectorType vectorType = readOp.getResult().getType().cast<VectorType>();
+  unsigned lanes = getVectorLaneSize(vectorType);
+
+  AffineExpr linearAccess = constructLinearizedAffineExpr(readOp, state);
+  if (linearAccess.isSymbolicOrConstant()) {
+    return success();
+  }
+
+  MemRefType memRefType = readOp.getSource().getType().cast<MemRefType>();
+  MLIRContext *context = memRefType.getContext();
+  ArrayRef<int64_t> sizes = memRefType.getShape();
+  int numDims = sizes.size();
+
+  auto block = readOp->getBlock();
+  assert(state->blockToEnclosingLoops.count(block) &&
+         "enclosing loops should have been computed for the read operation\n");
+  auto enclosingLoops = state->blockToEnclosingLoops[block];
+
+  SmallVector<Value, 4> indices(readOp.getIndices().begin(),
+                                readOp.getIndices().end());
+
+  // If the lowest dim has iv, check whether its corresponding loop step is
+  // divisible by the vector lanes.
+  int32_t step = 0;
+  if (auto dimExpr =
+          getAffineDimExpr(numDims - 1, context).dyn_cast<AffineDimExpr>()) {
+    auto index = indices[dimExpr.getPosition()];
+    // Iterate over all enclosing loops, and find the one that is variant in
+    // index.
+    for (auto loop : enclosingLoops) {
+      AffineForOp affineForOp = cast<AffineForOp>(loop);
+      auto iv = affineForOp.getInductionVar();
+      auto invariants = getInvariantAccesses(iv, indices);
+
+      if (!invariants.count(index)) {
+        step = affineForOp.getStep();
+        if (step % lanes) {
+          LLVM_DEBUG(llvm::dbgs() << "\n\n"
+                                  << *readOp
+                                  << "'s lowest dim's loop step is not "
+                                     "divisible by the vector lanes.\n\n");
+          return failure();
```

Comment:
Change lines 2313-2317 to: 
```suggestion
return readOp->emitError() 
             << "Loop step of inner index of " 
            << readOp->getName() 
            << " is not divisible by number of vector lanes.";
```